hadrian/src/Settings/Builders/Ghc.hs

```diff
@@ -153,6 +153,7 @@ includeGhcArgs = do
             , cIncludeArgs
             , arg $      "-I" ++ root -/- generatedDir
             , arg $ "-optc-I" ++ root -/- generatedDir
+            , arg $      "-I" ++ root -/- "stage1/gmp/include"
```

Comment:
iirc you can choose between building with a bundled gmp and the system gmp somehow. Maybe switching from one to the other helps (no idea what the default is on macos). Otherwise maybe worth checking if GHC HEAD builds on macos and if so what we can steal from that.

Reply:
I'll start a build of GHC HEAD 🤞

Reply:
hm, it failed with
```
Command line: cc -std=c99 -Wall --target=x86_64-apple-darwin -E -MM -MG -MF _build/stage1/libraries/ghc-bignum/build/c/cbits/gmp_wrappers.o.d -MT _build/stage1/libraries/ghc-bignum/build/c/cbits/gmp_wrappers.o -Irts/include -I_build/stage1/lib -I_build/stage1/libraries/ghc-bignum/build -I_build/stage1/libraries/ghc-bignum/build/include/ -I_build/stage1/libraries/ghc-bignum/build/include -Ilibraries/ghc-bignum/include/ -Ilibraries/ghc-bignum/include -I/Users/moises.ackerman/git/ghc/_build/stage1/lib/x86_64-osx-ghc-9.3.20211104/rts-1.0.2/include -x c libraries/ghc-bignum/cbits/gmp_wrappers.c
===> Command failed with error code: 1
libraries/ghc-bignum/cbits/gmp_wrappers.c:30:3: error: __GNU_MP_VERSION not defined
# error __GNU_MP_VERSION not defined
  ^
libraries/ghc-bignum/cbits/gmp_wrappers.c:42:3: error: WORD_SIZE_IN_BITS != GMP_LIMB_BITS not supported
# error WORD_SIZE_IN_BITS != GMP_LIMB_BITS not supported
  ^
2 errors generated.
Command failed
Build failed.
```
which seems related. I'll try the `stage1/gmp/include` fix on HEAD to see what happens.

Reply:
Looking at the [docs](https://gitlab.haskell.org/ghc/ghc/-/wikis/building/hadrian#first-build) it seems like `./configure --with-intree-gmp` might be helpful.

Reply:
oh, thanks!